bigtable-grpc-interface/src/main/java/com/google/cloud/bigtable/grpc/OAuth2CredentialsInterceptor.java

```diff
@@ -0,0 +1,206 @@
+/*
+ * Copyright 2014 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.cloud.bigtable.grpc;
+
+import io.grpc.Call;
+import io.grpc.Channel;
+import io.grpc.ClientInterceptor;
+import io.grpc.ClientInterceptors.CheckedForwardingCall;
+import io.grpc.Metadata;
+import io.grpc.MethodDescriptor;
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.util.concurrent.Future;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicReference;
+
+import com.google.api.client.util.Clock;
+import com.google.api.client.util.Preconditions;
+import com.google.auth.oauth2.AccessToken;
+import com.google.auth.oauth2.OAuth2Credentials;
+
+/**
+ * Client interceptor that authenticates all calls by binding header data provided by a credential.
+ * Typically this will populate the Authorization header but other headers may also be filled out.
+ *
+ * <p> Uses the new and simplified Google auth library:
+ * https://github.com/google/google-auth-library-java</p>
+ *
+ * <p> TODO: COPIED FROM io.grpc.auth.ClientAuthInterceptor.  The logic added here for
+ * initialization and locking could be moved back to gRPC.  This implementation takes advantage
+ * of the fact that all of the Bigtable endpoints are OAuth2 based.  It uses the OAuth AccessToken
+ * to get the token value and next refresh time.  The refresh is scheduled asynchronously.</p>
+ */
+public class OAuth2CredentialsInterceptor implements ClientInterceptor, Closeable {
+
+  private static final int ASYNC_TOKEN_MILLISECONDS = 60000;
+  private static final int REFRESH_WAIT_MS = 5000;
```

Comment:
MS or MILLISECONDS?
I think in some other places we have used 60 * 1000 and 5 * 1000?

I dont feel strongly about either but we should be consistent in both.